src/types.rs

```diff
@@ -14,11 +15,11 @@ use std::rc::Rc;
 pub trait TypeValidator {
     /// If the specified value violates one or more of this type's constraints,
     /// returns `false`, otherwise `true`
-    fn is_valid(&self, value: &OwnedElement, type_store: &TypeStore) -> bool;
+    fn is_valid(&self, value: &IonSchemaElement, type_store: &TypeStore) -> bool;
```

Comment:
question—Do you think it's worth exposing `IonSchemaElement` publicly here? Could we have two different functions (like this example), both of which convert to `IonSchemaElement` behind the scenes?
```rust
fn is_valid(&self, value: &OwnedElement, type_store: &TypeStore) -> bool;
fn is_valid(&self, value: &Vec<OwnedElement>, type_store: &TypeStore) -> bool;
```

Which option do you think is nicer for someone starting to use this library for the first time?

Reply:
This is an internal method and would not be exposed to user. The method that will exposed to user is:
```rust
     pub fn validate<I: Into<IonSchemaElement>>(&self, value: I) -> ValidationResult {
        ...
     }
```

This means that people can just pass on `&OwnedElement` and `&Vec<OwnedElement>` to this validate method and method takes care of the conversion to `IonSchemaElement`.

Currently it didn't have an implementation of `From<&Vec<OwnedElement>> for IonSchemaElement` but I will add it in following commit along with your suggested changes..

Reply:
🤦‍♂️ Right, we talked about this already, and I just forgot about it.